chain/types/blockheader.go

```diff
@@ -162,29 +172,31 @@ func CidArrsEqual(a, b []cid.Cid) bool {
 
 var blocksPerEpoch = NewInt(build.BlocksPerEpoch)
 
-func PowerCmp(eproof ElectionProof, mpow, totpow BigInt) bool {
+func IsTicketWinner(partialTicket []byte, ssizeI uint64, totpow BigInt, sampleRate int64) bool {
+	ssize := NewInt(ssizeI)
 
 	/*
 		Need to check that
-		(h(vrfout) + 1) / (max(h) + 1) <= e * minerPower / totalPower
+		(h(vrfout) + 1) / (max(h) + 1) <= e * sectorSize / totalPower
 		max(h) == 2^256-1
 		which in terms of integer math means:
-		(h(vrfout) + 1) * totalPower <= e * minerPower * 2^256
+		(h(vrfout) + 1) * totalPower <= e * sectorSize * 2^256
 		in 2^256 space, it is equivalent to:
-		h(vrfout) * totalPower < e * minerPower * 2^256
+		h(vrfout) * totalPower < e * sectorSize * 2^256
 	*/
 
-	h := sha256.Sum256(eproof)
+	h := sha256.Sum256(partialTicket)
 
 	lhs := BigFromBytes(h[:]).Int
 	lhs = lhs.Mul(lhs, totpow.Int)
+	lhs = lhs.Mul(lhs, big.NewInt(sampleRate))
```

Comment:
Shouldn't sample rate increase the probability by the `sampleRate`? Currently it decreases probability of winning.

Reply:
So, i think we're using this wrong. What gets passed in is the number of challenges for the miner being called, which is totalsectors / 25 or something

Reply:
Great catch. Yes, we want this:
`challenge ticket hash <? sectorPower / totalPower * 1/sectorSampled * ec.ExpectedLeaders

Accordingly, sectorsSampled_denom (defined as `sampleRate` here) should be on rhs, not lhs

Reply:
but yes, what should get passed in is the sample rate, not the num of tickets. specifically a miner with 10x more tickets than another should be able to sample 10x more tickets (even if both only sample eg 4%)